packages/discord.js/src/managers/GuildChannelManager.js

```diff
@@ -308,57 +309,78 @@ class GuildChannelManager extends CachedManager {
+          permissionOverwrites = newParent.permissionOverwrites.cache.map(overwrite =>
+            PermissionOverwrites.resolve(overwrite, this.guild),
+          );
+        }
+      } else {
+        const resolvedChannel = this.resolve(channel);
+        if (!resolvedChannel) throw new DiscordjsTypeError(ErrorCodes.InvalidType, 'channel', 'GuildChannelResolvable');
+        if (resolvedChannel.parent) {
+          permissionOverwrites = resolvedChannel.parent.permissionOverwrites.cache.map(overwrite =>
             PermissionOverwrites.resolve(overwrite, this.guild),
           );
         }
-      } else if (resolvedChannel.parent) {
-        permission_overwrites = resolvedChannel.parent.permissionOverwrites.cache.map(overwrite =>
-          PermissionOverwrites.resolve(overwrite, this.guild),
-        );
       }
     }
 
-    const newData = await this.client.rest.patch(Routes.channel(resolvedChannel.id), {
-      body: {
-        name: options.name,
-        type: options.type,
-        topic: options.topic,
-        nsfw: options.nsfw,
-        bitrate: options.bitrate,
-        user_limit: options.userLimit,
-        rtc_region: options.rtcRegion,
-        video_quality_mode: options.videoQualityMode,
-        parent_id: parentId,
-        lock_permissions: options.lockPermissions,
-        rate_limit_per_user: options.rateLimitPerUser,
-        default_auto_archive_duration: options.defaultAutoArchiveDuration,
-        permission_overwrites,
-        available_tags: options.availableTags?.map(availableTag => transformGuildForumTag(availableTag)),
-        default_reaction_emoji:
-          options.defaultReactionEmoji && transformGuildDefaultReaction(options.defaultReactionEmoji),
-        default_thread_rate_limit_per_user: options.defaultThreadRateLimitPerUser,
-        flags: 'flags' in options ? ChannelFlagsBitField.resolve(options.flags) : undefined,
-        default_sort_order: options.defaultSortOrder,
-        default_forum_layout: options.defaultForumLayout,
+    const snakeCaseBody = Object.assign(
+      // Remove properties we don't want to pass to the API in the body
+      {
+        set parent(_) {},
+        set position(_) {},
+        set reason(_) {},
       },
+      toSnakeCase(options),
+    );
+
+    // This overwrites a passed snake_case version if a camelCase version OR `parent` is passed
+    if (parentId) {
+      snakeCaseBody.parent_id = parentId;
+    }
+
+    // This overwrites a passed snake_case version if a camelCase version is passed
+    if (permissionOverwrites) {
+      snakeCaseBody.permission_overwrites = permissionOverwrites;
+    }
+
+    if (snakeCaseBody.available_tags) {
+      snakeCaseBody.available_tags = snakeCaseBody.available_tags.map(availableTag =>
+        'emoji' in availableTag ? transformGuildForumTag(availableTag) : availableTag,
+      );
+    }
+
+    if (snakeCaseBody.default_reaction_emoji?.id) {
```

Comment:
Any reason for checking `id` specifically here? Passing `name` wouldn't work this way.
```suggestion
    if (snakeCaseBody.default_reaction_emoji) {
```

If there is a reason for the `id` check then we must also check for `name`.

Reply:
I think we should allow it (to better support new features that don't have real support yet), but I don't know that we should actually document it that way. It might be something we mention somewhere once, but I don't think we should specifically call it out except *maybe* types...my opinion will probably change once the lib is actually ts native and the underlying support for raw data is way better.